client/src/components/SignupForm.jsx

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import {
+  Form, Input, Button, message,
+} from 'antd';
+import { Link } from 'react-router-dom';
+import http from '../services/httpService';
+import Img from './common/Img';
+import ParagraphText from './common/Paragraph';
+import HeaderTitle from './common/Title';
+import './signup.css';
+
+const postData = async (userInfo) => {
+  try {
+    await http.post('/api/v1/signup', userInfo);
```

Comment:
I think we need to setUser to userData, So we'll need to get user data from the server and add setUser to auth provider.
then change this function to receive callbak wich will setUser to user data and then `history.push('/')`